### ISSUE TITLE
TerriaJS as a CKAN previewer

### ISSUE THREAD

**Issue by rsignell-usgs** [ioos/catalog-ckan#141]
Would it be cool to implement this?
http://terria.io/Documentation/guide/deploying/using-as-a-ckan-previewer/
TerriaJS is used for the map viewer at data.gov.au (which also uses CKAN)

**Comment by rsignell-usgs** [comment]
Here's an example: https://data.gov.au/dataset/taxation-statistics-2013-14/resource/9f10b598-fbbb-4028-a982-80d317de7c34

![2017-02-15_15-04-53](https://cloud.githubusercontent.com/assets/1872600/22993193/24b582be-f390-11e6-88d4-30ec094b77cb.png)

**Comment by rsignell-usgs** [comment]
The AmeriGEOS program has made nice use of the CKAN API to access data from a variety of sources:
https://data.amerigeoss.org/ameriterria/
(go to "add data" and select one of the listed CKAN endpoints to make the query, then expand the "No Group" to see the datasets returned)  
Here's an example of a [WMS layer returned from the CKAN query for Peru](https://data.amerigeoss.org/ameriterria/#start=%7B%22version%22%3A%220.0.05%22%2C%22initSources%22%3A%5B%7B%22initFragment%22%3A%22datahub_ar%22%7D%2C%7B%22initFragment%22%3A%22datahub_pe%22%7D%2C%7B%22initFragment%22%3A%22datahub_ioos%22%7D%2C%7B%22initFragment%22%3A%22datahub_unhdx%22%7D%2C%7B%22initFragment%22%3A%22datahub_carigeo%22%7D%2C%7B%22initFragment%22%3A%22datahub_ciifen%22%7D%2C%7B%22initFragment%22%3A%22datahub_ca%22%7D%2C%7B%22initFragment%22%3A%22datahub_br%22%7D%2C%7B%22initFragment%22%3A%22datahub_geosur%22%7D%2C%7B%22initFragment%22%3A%22datahub_mx%22%7D%2C%7B%22initFragment%22%3A%22datahub_world-bank%22%7D%2C%7B%22initFragment%22%3A%22datahub_caribbean-marine-atlas-cma%22%7D%2C%7B%22initFragment%22%3A%22datahub_amerigeo%22%7D%2C%7B%22initFragment%22%3A%22datahub_amerigeo-arcgis%22%7D%2C%7B%22initFragment%22%3A%22datahub_esri%22%7D%2C%7B%22initFragment%22%3A%22datahub_servir%22%7D%2C%7B%22initFragment%22%3A%22datahub_wfp%22%7D%2C%7B%22initFragment%22%3A%22datahub_unescoihpwins%22%7D%2C%7B%22initFragment%22%3A%22datahub_gfdrr%22%7D%2C%7B%22initFragment%22%3A%22datahub_us%22%7D%2C%7B%22catalog%22%3A%5B%7B%22name%22%3A%22User-Added+Data%22%2C%22description%22%3A%22The+group+for+data+that+was+added+by+the+user+via+the+Add+Data+panel.%22%2C%22info%22%3A%5B%5D%2C%22isUserSupplied%22%3Atrue%2C%22isPromoted%22%3Afalse%2C%22isHidden%22%3Afalse%2C%22forceProxy%22%3Afalse%2C%22customProperties%22%3A%7B%7D%2C%22id%22%3A%22Root+Group%2FUser-Added+Data%22%2C%22shortReportSections%22%3A%5B%5D%2C%22isWaitingForDisclaimer%22%3Afalse%2C%22hideSource%22%3Afalse%2C%22nameInCatalog%22%3A%22User-Added+Data%22%2C%22isOpen%22%3Afalse%2C%22items%22%3A%5B%5D%2C%22preserveOrder%22%3Afalse%2C%22type%22%3A%22group%22%2C%22parents%22%3A%5B%5D%7D%5D%7D%2C%7B%22sharedCatalogMembers%22%3A%7B%22Root+Group%2FAmerigeoss+Peru%22%3A%7B%22isOpen%22%3Atrue%2C%22type%22%3A%22ckan%22%2C%22parents%22%3A%5B%5D%7D%2C%22Root+Group%2FAmerigeoss+Peru%2F_ungrouped%22%3A%7B%22isOpen%22%3Atrue%2C%22type%22%3A%22group%22%2C%22parents%22%3A%5B%22Root+Group%2FAmerigeoss+Peru%22%5D%7D%2C%22Root+Group%2FAmerigeoss+Peru%2F_ungrouped%2F2243f699-8667-4c2d-92e0-65986f6ba338%22%3A%7B%22nowViewingIndex%22%3A0%2C%22isEnabled%22%3Atrue%2C%22isShown%22%3Atrue%2C%22isLegendVisible%22%3Atrue%2C%22useOwnClock%22%3Afalse%2C%22opacity%22%3A0.6%2C%22keepOnTop%22%3Afalse%2C%22splitDirection%22%3A0%2C%22showOnChart%22%3Afalse%2C%22type%22%3A%22esri-mapServer%22%2C%22parents%22%3A%5B%22Root+Group%2FAmerigeoss+Peru%22%2C%22Root+Group%2FAmerigeoss+Peru%2F_ungrouped%22%5D%7D%7D%7D%2C%7B%22initialCamera%22%3A%7B%22west%22%3A-88.70863349468657%2C%22south%22%3A-17.457011039834157%2C%22east%22%3A-61.36870664041641%2C%22north%22%3A-1.5900310332198935%2C%22position%22%3A%7B%22x%22%3A2280020.592666619%2C%22y%22%3A-8532182.739049414%2C%22z%22%3A-1517263.6623314144%7D%2C%22direction%22%3A%7B%22x%22%3A-0.2543901348080352%2C%22y%22%3A0.9519664533621764%2C%22z%22%3A0.17042749480464933%7D%2C%22up%22%3A%7B%22x%22%3A0.043998765672747936%2C%22y%22%3A-0.16465005194248666%2C%22z%22%3A0.9853702192651304%7D%7D%2C%22homeCamera%22%3A%7B%22west%22%3A-121%2C%22south%22%3A18%2C%22east%22%3A-66%2C%22north%22%3A61%7D%2C%22baseMapName%22%3A%22Bing+Maps+Aerial+with+Labels%22%2C%22viewerMode%22%3A%223d%22%2C%22currentTime%22%3A%7B%22dayNumber%22%3A2459081%2C%22secondsOfDay%22%3A4833.41%7D%2C%22showSplitter%22%3Afalse%2C%22splitPosition%22%3A0.5%7D%2C%7B%22stories%22%3A%5B%5D%7D%5D%7D)


Has anyone tried generating CKAN catalog entries that allow exploration of the IOOS Catatog CKAN endpoint?

**Comment by mwengren** [comment]
This looks like the correct CKAN/TerriaJS plugin: https://github.com/TerriaJS/ckanext-terria_view.  

Immediate task is to look into implementing this to allow embedded preview of compatible services (WMS) in IOOS Catalog. 

@rsignell-usgs we should probably separate out your previous https://github.com/ioos/catalog-ckan/issues/141#issuecomment-676413319 into a separate issue since it's more about either deploying TerriaJS for IOOS Catalog data or investigating compatibility with AmeriTerria, right?

**Comment by rsignell-usgs** [comment]
Yes @mwengren , sorry, I will raise a new issue on the Terria/WMS issue.   Also, if we need help getting this going, the USGS contractor who set up the Amerigeoss CKAN catalog with TerriaJS previewer said we could do a screenshare and he could help, as he just went through it! 
 
https://data.amerigeoss.org/dataset/coawst-forecast-system-usgs-us-east-coast-and-gulf-of-mexico-experimental/resource/2fa4bb60-0acd-4460-a612-df8a1b019af0

![2020-09-14_11-17-07](https://user-images.githubusercontent.com/1872600/93104861-6212d380-f67c-11ea-87d3-2dfa3a77da00.png)
![2020-09-14_11-16-04](https://user-images.githubusercontent.com/1872600/93104880-66d78780-f67c-11ea-957d-a61b715203db.jpg)

**Comment by benjwadams** [comment]
I'm set to start looking at this.  The directions given in the original link seem geared towards the Australian National Map, so I am going to want to know how to swap out the map location to something more centered on the Americas.

**Comment by greenteadigital** [comment]
Can someone assign this to me pls?

**Comment by greenteadigital** [comment]
The `ckanext-terria_view` plugin is dependent on Terria v7 (deprecated), written in Python 2, and requires CKAN v2.7.
Given that all of that software is outdated, I'm hoping to find a way to make this happen with Terria v8, Python 3, and CKAN v2.9

**Comment by benjwadams** [comment]
Closing due to extension being unsupported in current CKAN versions.
